operate/webapp/src/main/java/io/camunda/operate/webapp/security/permission/PermissionsService.java

```diff
@@ -0,0 +1,287 @@
+import io.camunda.operate.property.OperateProperties;
+import io.camunda.operate.webapp.security.identity.IdentityPermission;
+import io.camunda.security.auth.Authorization;
+import io.camunda.security.auth.SecurityContext;
+import io.camunda.security.impl.AuthorizationChecker;
+import io.camunda.zeebe.protocol.record.value.AuthorizationResourceType;
+import io.camunda.zeebe.protocol.record.value.PermissionType;
+import jakarta.annotation.PostConstruct;
+import java.util.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+public class PermissionsService {
+
+  public static final String RESOURCE_KEY_ALL = "*";
+  public static final String RESOURCE_TYPE_PROCESS_DEFINITION = "process-definition";
+  public static final String RESOURCE_TYPE_DECISION_DEFINITION = "decision-definition";
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PermissionsService.class);
+
+  private final OperateProperties operateProperties;
+  private final AuthorizationChecker authorizationChecker;
+  private final TenantsService tenantsService;
+
+  public PermissionsService(
+      final OperateProperties operateProperties,
+      final AuthorizationChecker authorizationChecker,
+      final TenantsService tenantsService) {
+    this.operateProperties = operateProperties;
+    this.authorizationChecker = authorizationChecker;
+    this.tenantsService = tenantsService;
+  }
+
+  @PostConstruct
+  public void logCreated() {
+    LOGGER.debug("PermissionsService bean created.");
+  }
+
+  /**
+   * getProcessDefinitionPermissions
+   *
+   * @param bpmnProcessId bpmnProcessId
+   * @return permissions the user has for the given bpmnProcessId
+   */
+  public Set<String> getProcessDefinitionPermissions(final String bpmnProcessId) {
+    return getResourcePermissions(bpmnProcessId, AuthorizationResourceType.PROCESS_DEFINITION);
+  }
+
+  /**
+   * getDecisionDefinitionPermissions
+   *
+   * @param decisionId decisionId
+   * @return permissions the user has for the given decisionId
+   */
+  public Set<String> getDecisionDefinitionPermissions(final String decisionId) {
+    return getResourcePermissions(decisionId, AuthorizationResourceType.DECISION_DEFINITION);
+  }
+
+  /**
+   * getResourcePermissions
+   *
+   * @param resourceId resourceId
+   * @param resourceType resourceType
+   * @return permissions the user has for the given resource
+   */
+  public Set<String> getResourcePermissions(
+      final String resourceId, final AuthorizationResourceType resourceType) {
+    final Set<String> permissions = new HashSet<>();
+    if (permissionsEnabled() && isAuthenticated()) {
+      for (PermissionType permissionType : PermissionType.values()) {
+        if (isAuthorized(resourceId, resourceType, permissionType)) {
+          permissions.add(permissionType.name());
+        }
+      }
```

Comment:
⚠️ This loop will currently trigger 14 queries to database
Why not adding a new method in AuthorizationChecker that does the same in one query?